modules/new_profile/php/new_profile.class.inc

```diff
@@ -44,19 +53,278 @@ class New_Profile extends \NDB_Page
         return false;
     }
     /**
-     * Include additional JS files:
-     *  1. editForm.js - reactified form to update media
+     * Processes the values and saves to database
+     *
+     * @param array $values form values
+     *
+     * @return void
+     */
+    function _process($values): void
+    {
+        // Change empty strings to null before passing as arguments.
+        foreach ($values as $key => $value) {
+            if (empty($value)) {
+                $values[$key] = null;
+            }
+        }
+        /* When a user has only one site, or when a centerID has not been
+         * passed, choose the first one in the User's centerID list.
+         */
+        $centerIDs = \User::singleton()->getData('CenterIDs');
+        if (count($centerIDs) <= 1 || !isset($values['site'])) {
+            $values['site'] = $centerIDs[0];
+        }
+
+        /* Create the candidate and retrieve its ID.
+         * Use form values when present, otherwise default to null.
+         */
+        $candID = \Candidate::createNew(
+            (int)$values['site'],
+            $values['dobDate'],
+            $values['edcDate'] ?? null,
+            $values['sex'],
+            $values['pscid'] ?? null,
+            isset($values['project']) ? intval($values['project']) : null
+        );
+
+        /* Update front-end data. Use passed PSCID when present, otherwise
+         * retrieve it from the newly-created candidate.
+         */
+        $this->candID  = $candID;
+        $this->pscid   = $values['pscid']
+            ?? \Candidate::singleton($candID)->getPSCID();
+
+        // freeze it, just in case
+        $this->form->freeze();
+    }
+
+    /**
+     * This function will return a json file with DCCID and PSCID
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $this->setup();
+        if ($request->getMethod() === "POST") {
+            $this->save();
```

Comment:
Since you're overriding handle, did you test this both with and without valid permissions?

Reply:
yep, you can't access without data_entry permissions